incubator/virtualcluster/pkg/vn-agent/server/route.go

```diff
@@ -119,90 +114,43 @@ func (s *Server) InstallHandlers() {
 
-		// there must be a peer certificate in the tls connection
-		if req.Request.TLS == nil || len(req.Request.TLS.PeerCertificates) == 0 {
-			resp.ResponseWriter.WriteHeader(http.StatusForbidden)
-			return
-		}
 		tenantName := req.Request.TLS.PeerCertificates[0].Subject.CommonName
 		TranslatePath(req, tenantName)
 
 		klog.V(4).Infof("request after translate %+v", req.Request.URL)
-
-		transport = &http.Transport{
-			TLSClientConfig: &tls.Config{
-				InsecureSkipVerify: true,
-				Certificates:       []tls.Certificate{*s.config.KubeletClientCert},
-			},
-		}
 	} else {
 		klog.Info("will forward request to super apiserver")
 		// forward request to super apiserver
-		if req.Request.TLS == nil || len(req.Request.TLS.PeerCertificates) == 0 {
-			resp.ResponseWriter.WriteHeader(http.StatusForbidden)
-			return
-		}
-		// 1. get incluster rest config
-		restConfig, err := rest.InClusterConfig()
-		if err != nil {
-			klog.Errorf("fail to sa token or root ca: %s", err)
-			resp.ResponseWriter.WriteHeader(http.StatusNotFound)
-			resp.ResponseWriter.Write([]byte(err.Error()))
-			return
-		}
-		// 1. get super master address
-		superHttpsUrl, err := url.Parse(restConfig.Host)
-		if err != nil {
-			klog.Errorf("fail to get super apiserver's url: %s", err)
-			resp.ResponseWriter.WriteHeader(http.StatusNotFound)
-			resp.ResponseWriter.Write([]byte(err.Error()))
-			return
-		}
-
-		// 2. As we use the sa of the vn-agent, to forward the request, make sure
-		// the sa has the permission to access the pods
 		tenantName := req.Request.TLS.PeerCertificates[0].Subject.CommonName
-		err = TranslatePathForSuper(req, tenantName)
+		err := TranslatePathForSuper(req, tenantName)
 		if err != nil {
 			klog.Errorf("fail to translate url path for super master: %s", err)
 			resp.ResponseWriter.WriteHeader(http.StatusNotFound)
 			resp.ResponseWriter.Write([]byte(err.Error()))
-		}
-		klog.V(4).Infof("request after translate %+v", superHttpsUrl)
-
-		// 3. mutate the request, i.e., replacing the dst, add bearer token header
-		req.Request.URL.Host = superHttpsUrl.Host
-		req.Request.URL.Scheme = "https"
-		req.Request.Header.Add("Authorization", "Bearer "+restConfig.BearerToken)
-		if err != nil {
-			resp.ResponseWriter.WriteHeader(http.StatusNotFound)
-			resp.ResponseWriter.Write([]byte(err.Error()))
-			return
-		}
-
-		caCrtPool, err := certutil.NewPool(restConfig.TLSClientConfig.CAFile)
-		if err != nil {
-			resp.ResponseWriter.WriteHeader(http.StatusNotFound)
-			resp.ResponseWriter.Write([]byte(err.Error()))
 			return
 		}
+		klog.V(4).Infof("request after translate %+v", req.Request.URL)
 
-		transport = &http.Transport{
-			TLSClientConfig: &tls.Config{
-				RootCAs: caCrtPool,
-			},
-		}
+		// mutate the request, i.e., replacing the dst, add bearer token header
+		req.Request.URL.Host = s.superAPIServerAddress.Host
+		req.Request.URL.Scheme = "https"
+		req.Request.Header.Add("Authorization", "Bearer "+s.restConfig.BearerToken)
```

Comment:
Do all auth providers use bearer tokens for auth?

Reply:
The token is got from the InClusterConfig generated from the Pod service account/secret. Hence I assume it is valid in most of the cases.

Reply:
@Fei-Guo awesome, thanks!

Reply:
Bearer token is not the only way used by Kubernetes to authenticate HTTP requests, but it should work here. Do you mean super apiserver may disable the bearer-token authentication?